hummingbot/connector/exchange/paper_trade/paper_trade_exchange.pyx

```diff
@@ -484,51 +513,76 @@ cdef class PaperTradeExchange(ExchangeBase):
                                    order_id,
                                    base_asset,
                                    quote_asset,
-                                   base_asset if config.buy_fees_asset is AssetType.BASE_CURRENCY else quote_asset,
-                                   total_base_acquired,
-                                   total_quote_needed,
-                                   s_decimal_0,
+                                   base_asset,
+                                   acquired_amount,
+                                   sold_amount,
+                                   fee_amount,
                                    OrderType.MARKET))
 
     cdef c_execute_sell(self, str order_id, str trading_pair_str, object amount):
         cdef:
-            object quote_asset_amount
-            object base_asset_amount
-        config = self._config
-        quote_asset = self._trading_pairs[trading_pair_str].quote_asset
-        quote_asset_amount = self.c_get_balance(quote_asset)
-        base_asset = self._trading_pairs[trading_pair_str].base_asset
-        base_asset_amount = self.c_get_balance(base_asset)
-
-        if amount > base_asset_amount:
+            str quote_asset = self._trading_pairs[trading_pair_str].quote_asset
+            str base_asset = self._trading_pairs[trading_pair_str].base_asset
+            object quote_balance = self.c_get_balance(quote_asset)
+            object base_balance = self.c_get_balance(base_asset)
+
+        order_book = self.order_books[trading_pair_str]
+
+        sell_entries = order_book.simulate_sell(amount)
+
+        # Get the weighted average price of the trade
+        avg_price = Decimal(0)
+        for entry in sell_entries:
+            avg_price += Decimal(entry.price) * Decimal(entry.amount)
+        avg_price = avg_price / amount
+
+        order_candidate = OrderCandidate(
+            trading_pair=trading_pair_str,
+            # Market orders are not maker orders
+            is_maker=False,
+            order_type=OrderType.MARKET,
+            order_side=TradeType.SELL,
+            amount=amount,
+            price=avg_price
+        )
+
+        adjusted_candidate_order = self._budget_checker.adjust_candidate(order_candidate, all_or_none=True)
+
+        # Base currency used, including fees.
+        sold_amount = adjusted_candidate_order.order_collateral.amount
+        # Quote currency acquired, including fees.
+        acquired_amount = adjusted_candidate_order.potential_returns.amount
+
+        fee_amount = adjusted_candidate_order.collateral_dict[base_asset]
```

Comment:
Same question here. Is it ok to assume we always pay the fee in the base asset? If that is not the case we could be getting the wrong fee amount.